cmd/service.go

```diff
@@ -66,12 +82,40 @@ A full list of service types that can be deployed are available using: 'odo cata
 			fmt.Printf("%s service already exists in the current application.\n", serviceName)
 			os.Exit(1)
 		}
-		err = svc.CreateService(client, serviceName, serviceType, applicationName)
+		err = svc.CreateService(client, serviceName, serviceType, servicePlan, parameters, applicationName)
 		checkError(err, "")
 		fmt.Printf("Service '%s' was created.\n", serviceName)
 	},
 }
 
+var serviceLinkCmd = &cobra.Command{
+	Use:   "link <secret> <component_name>",
```

Comment:
> so how can user link them to the service?

You don't have to create a secret when a `serviceInstance` is created as the service catalog will populate a secret within the current namespace.

Reply:
> odo link

This command doesn't support for the moment to add the secret's info to the DC as such

```
	// Add the Secret as EnvVar to the container
	dc.Spec.Template.Spec.Containers[0].EnvFrom = []corev1.EnvFromSource{
		{
			SecretRef: &corev1.SecretEnvSource{
				LocalObjectReference: corev1.LocalObjectReference{Name: secretName},
			},
		},
	}
```

Reply:
I've just summarized my thoughts on linking here https://github.com/redhat-developer/odo/issues/652#issuecomment-424996402

Reply:
@kadel since I am working on getting this PR to a state where we can merge it, is this change (and the way that `odo link` is meant to work) decided? Should I move ahead with the change?